connectors/golden-tables/src/test/scala/io/delta/golden/GoldenTables.scala

```diff
@@ -864,6 +864,97 @@ class GoldenTables extends QueryTest with SharedSparkSession {
     }
   }
   */
+
+  def writeBasicDecimalTable(tablePath: String): Unit = {
+    val data = Seq(
+      Seq("234", "1", "2", "3"),
+      Seq("2342222.23454", "111.11", "22222.22222", "3333333333.3333333333"),
+      Seq("0.00004", "0.001", "0.000002", "0.00000000003"),
+      Seq("-2342342.23423", "-999.99", "-99999.99999", "-9999999999.9999999999")
+    ).map(_.map(new JBigDecimal(_))).map(Row(_: _*))
+
+    val schema = new StructType()
+      .add("part", new DecimalType(12, 5))
+      .add("col1", new DecimalType(5, 2))
+      .add("col2", new DecimalType(10, 5))
+      .add("col3", new DecimalType(20, 10))
+
+    spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+      .repartition(1)
+      .write
+      .format("delta")
+      .partitionBy("part")
+      .save(tablePath)
+  }
+
+  generateGoldenTable("basic-decimal-table") { tablePath =>
+    writeBasicDecimalTable(tablePath)
+  }
+
+  generateGoldenTable("basic-decimal-table-legacy") { tablePath =>
+    withSQLConf(("spark.sql.parquet.writeLegacyFormat", "true")) {
+      writeBasicDecimalTable(tablePath)
+    }
+  }
+
+  // todo: how to have dictionary encoding for fixed_len_byte_array?
```

Comment:
Not sure how to do this, for the others having low cardinality for the set of values seemed to work.

Dictionary support for fixed_len_byte_array/binary is implemented in Spark, but we can just not support it if cannot find a way to test it.